[18.0][MIG] hr_timesheet_portal: Migration to 18.0

innovara opened the pull request:
This migration follows the recent migration of `hr_timesheet_portal` to 17.0. There is already [OCA/timesheet#800](https://github.com/OCA/timesheet/pull/800) for 18.0, but that one is more of a direct drop-in of code that works in 18.0, whereas this PR includes the full commit history and therefore provides traceability of the changes. However, the contribution of its author is acknowledged, as they were added to the list of contributors in the migration to 17.0.

Speaking of which:

**hr_timesheet_portal/static/src/js/hr_timesheet_portal.esm.js**  
- Replaced `session` and bound `user` service with a static `user` import.  
- Used `browser.setTimeout` instead of the global `setTimeout` to satisfy ESLint.  
- Removed legacy `this.user` binding and unused imports.  

**hr_timesheet_portal/templates/portal.xml**  
- Replaced `analytic_account_id` with `account_id`.

@hbrunn once again, would you be so kind to review it for me please?

hbrunn committed:
[ADD] hr_timesheet_portal

alexey-pelykh committed:
Apply dotfiles

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: timesheet-12.0/timesheet-12.0-hr_timesheet_portal
Translate-URL: https://translation.odoo-community.org/projects/timesheet-12-0/timesheet-12-0-hr_timesheet_portal/it/

hbrunn committed:
[IMP] hr_timesheet_portal: pre-commit auto fixes

hbrunn committed:
[MIG] hr_timesheet_portal to v14

hbrunn committed:
[ADD] allow portal users with timesheet group to be linked to employees

innovara committed:
[MIG] hr_timesheet_portal: Migration to 15.0

Signed-off-by: Manuel Fombuena <mfombuena@innovara.tech>

innovara committed:
[MIG] hr_timesheet_portal: Migration to 16.0

Signed-off-by: Manuel Fombuena <mfombuena@innovara.tech>

innovara committed:
[IMP] hr_timesheet_portal: pre-commit auto fixes

innovara committed:
[MIG] hr_timesheet_portal: Migration to 17.0

innovara committed:
[IMP] hr_timesheet_portal: add USAGE.md

Add USAGE.md and screenshots to improve documentation.

Signed-off-by: Manuel Fombuena <mfombuena@innovara.tech>

innovara committed:
[IMP] hr_timesheet_portal: pre-commit auto fixes

innovara committed:
[MIG] hr_timesheet_portal: Migration to 18.0

innovara commented:
@CRogos @BhaveshHeliconia you kindly reviewed the previous migration of this module to `17.0`, so I would really appreciate it if you could also take a look at this one when you have a moment.

CRogos reviewed (approved):
LGTM, but shoudn't we add a suitable user group automatically to the editor group?

maybe hr_timesheet.group_hr_timesheet_user but at least base.user_admin

innovara committed:
[IMP] hr_timesheet_portal: add admin to portal edit group and update syntax

- Add inheritance from group_hr_timesheet_portal to base.group_system.
- Update security XML and record rules to Odoo 18 Command syntax.

Signed-off-by: Manuel Fombuena <mfombuena@innovara.tech>

yannoliv reviewed (approved):
I have tested it on my local machine. Migration wise this looks OK!

hbrunn reviewed (approved):
/ocabot merge nobump
